src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(injuredAnimal) {
+    if (injuredAnimal instanceof Herbivore && !injuredAnimal.hidden) {
+      injuredAnimal.health -= 50;
+    }
+
+    if (!injuredAnimal.health) {
+      const listPosition = Animal.alive.indexOf(injuredAnimal);
+
+      (listPosition !== -1) && Animal.alive.splice(listPosition, 1);
+    }
+  }
```

Comment:
```suggestion
    if (!injuredAnimal.health) {
      const listPosition = Animal.alive.indexOf(injuredAnimal);

      Animal.alive.splice(listPosition, 1);
    }
  }
```

Reply:
@nzrii-kysel , yeah, I also thought to remove it, but the first point of the checklist confused me a bit, so I decided to leave it.

Reply:
my bad, forgot about the checklist, you were right not to remove it